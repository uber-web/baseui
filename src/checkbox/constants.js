// @flow
/* eslint-disable import/prefer-default-export */

export const STATE_TYPE = {
  change: 'CHANGE',
  focus: 'FOCUS',
  blur: 'BLUR',
  hover: 'HOVER',
  unhover: 'UNHOVER',
};

/* eslint-enable import/prefer-default-export */
