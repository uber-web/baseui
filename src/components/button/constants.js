/* eslint-disable flowtype/require-valid-file-annotation */

export const KIND = {
  square: 'square',
  round: 'round',
};

export const SIZE = {
  default: 'default',
  compact: 'compact',
};
