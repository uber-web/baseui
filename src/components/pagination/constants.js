/* eslint-disable flowtype/require-valid-file-annotation */
/* eslint-disable import/prefer-default-export */

export const STATE_CHANGE_TYPE = {
  prevPage: 'prevPage',
  nextPage: 'nextPage',
};
