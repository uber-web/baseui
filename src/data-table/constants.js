/*
Copyright (c) 2018-2020 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

export const COLUMNS = Object.freeze({
  ANCHOR: 'ANCHOR',
  CATEGORICAL: 'CATEGORICAL',
  NUMERICAL: 'NUMERICAL',
  BOOLEAN: 'BOOLEAN',
  STRING: 'STRING',
  CUSTOM: 'CUSTOM',
});

export const NUMERICAL_FORMATS = Object.freeze({
  DEFAULT: 'DEFAULT',
  ACCOUNTING: 'ACCOUNTING',
  PERCENTAGE: 'PERCENTAGE',
});

export const NUMERICAL_OPERATIONS = Object.freeze({
  EQ: 'EQ',
  GT: 'GT',
  GTE: 'GTE',
  LT: 'LT',
  LTE: 'LTE',
});

export const SORT_DIRECTIONS = Object.freeze({
  ASC: 'ASC',
  DESC: 'DESC',
});
