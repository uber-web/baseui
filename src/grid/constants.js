/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

export const BEHAVIOR = Object.freeze({
  fluid: 'fluid',
  fixed: 'fixed',
});

export const ALIGNMENT = Object.freeze({
  start: 'flex-start',
  center: 'center',
  end: 'flex-end',
});

export const DEFAULT_GRID_COLUMNS = [4, 8, 12];
export const DEFAULT_GRID_GUTTERS = [16, 36, 36];
export const DEFAULT_GRID_MARGINS = [16, 36, 64];
export const DEFAULT_GRID_MAX_WIDTH = 1280;
