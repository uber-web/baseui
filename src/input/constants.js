/* eslint-disable import/prefer-default-export */
export const STATE_TYPE = {
  change: 'change',
  focus: 'focus',
  blur: 'blur',
};
/* eslint-enable import/prefer-default-export */
