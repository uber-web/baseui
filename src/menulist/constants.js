// @flow
/* eslint-disable import/prefer-default-export */

export const STATE_CHANGE_TYPES = {
  keyPressArrowUp: 'keyPressArrowUp',
  keyPressArrowDown: 'keyPressArrowDown',
};

// Dict of props to prepend $ to pass thru to styletron
export const STYLETRON_PROP_MAPPER = {
  ref: true,
  isHighlighted: true,
};

export const KEY_STRINGS = {
  ArrowUp: 'ArrowUp',
  ArrowDown: 'ArrowDown',
  Enter: 'Enter',
};
