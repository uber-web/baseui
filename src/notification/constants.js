/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow

export const KIND = Object.freeze({
  primary: 'primary',
  success: 'success',
  warning: 'warning',
  error: 'error',
});
