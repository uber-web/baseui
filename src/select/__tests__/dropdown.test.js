/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import * as React from 'react';
import {mount} from 'enzyme';
import SelectDropdown from '../dropdown.js';
import {SIZE, TYPE} from '../constants.js';
import {StatefulMenu} from '../../menu/index.js';

jest.mock('../../menu');

describe('SelectDropdown', function() {
  let wrapper;
  let props = {};
  const options = [{id: '1', label: 'label1'}, {id: '2', label: 'label2'}];
  const value = [{id: '1', label: 'label1'}];
  const ref = React.createRef();

  beforeEach(function() {
    props = {
      value,
      valueKey: 'id',
      labelKey: 'label',
      size: SIZE.default,
      options,
      onItemSelect: jest.fn(),
      getOptionLabel: jest.fn(({option}) => <span>option.label</span>),
      maxDropdownHeight: '1000px',
      overrides: {},
      error: false,
      isLoading: false,
      multi: false,
      required: false,
      searchable: true,
      type: TYPE.select,
      width: 100,
    };
    wrapper = mount(<SelectDropdown innerRef={ref} {...props} />);
  });

  afterEach(function() {
    wrapper && wrapper.unmount();
  });

  afterAll(function() {
    jest.restoreAllMocks();
  });

  test('renders StatefulMenu', function() {
    // $FlowFixMe
    const menuProps = StatefulMenu.mock.calls[0][0];
    expect(StatefulMenu).toHaveBeenCalled();
    expect(menuProps.items).toMatchInlineSnapshot(`
Object {
  "__ungrouped": Array [
    Object {
      "id": "1",
      "label": "label1",
    },
    Object {
      "id": "2",
      "label": "label2",
    },
  ],
}
`);
    expect(menuProps.overrides).toMatchInlineSnapshot(`
Object {
  "List": Object {
    "component": Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [MockFunction],
    },
    "props": Object {
      "$maxHeight": "1000px",
      "aria-multiselectable": false,
    },
    "style": [Function],
  },
  "Option": Object {
    "props": Object {
      "getItemLabel": [Function],
      "onMouseDown": [Function],
      "overrides": Object {
        "ListItem": Object {
          "component": Object {
            "$$typeof": Symbol(react.forward_ref),
            "__STYLETRON__": undefined,
            "displayName": "StyledListItem",
            "render": [MockFunction],
          },
          "props": Object {
            "role": "option",
          },
          "style": undefined,
        },
      },
    },
  },
}
`);
  });

  test('StatefulMenu overrides merge with default overrides', function() {
    const emptyStateOverrides = {
      EmptyState: {
        style: {
          backgroundColor: 'red',
        },
      },
    };
    wrapper.setProps({
      overrides: {
        StatefulMenu: {
          props: {
            overrides: emptyStateOverrides,
          },
        },
      },
    });
    expect(wrapper.find(StatefulMenu).props().overrides.EmptyState)
      .toMatchInlineSnapshot(`
Object {
  "style": Object {
    "backgroundColor": "red",
  },
}
`);

    expect(wrapper.find(StatefulMenu).props().overrides.Option)
      .toMatchInlineSnapshot(`
Object {
  "props": Object {
    "getItemLabel": [Function],
    "onMouseDown": [Function],
    "overrides": Object {
      "ListItem": Object {
        "component": Object {
          "$$typeof": Symbol(react.forward_ref),
          "__STYLETRON__": undefined,
          "displayName": "StyledListItem",
          "render": [MockFunction],
        },
        "props": Object {
          "role": "option",
        },
        "style": undefined,
      },
    },
  },
}
`);
  });
});
