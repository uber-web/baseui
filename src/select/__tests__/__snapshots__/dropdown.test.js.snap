// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDropdown passes correct props to OptionContent for a selected item: OptionContent gets correct props when an option is selected 1`] = `
Object {
  "$disabled": undefined,
  "$error": false,
  "$isHighlighted": undefined,
  "$isLoading": false,
  "$multi": false,
  "$required": false,
  "$searchable": true,
  "$selected": true,
  "$size": "default",
  "$type": "select",
  "$width": 100,
  "aria-readonly": undefined,
  "aria-selected": true,
  "children": <span>
    option.label
  </span>,
}
`;

exports[`SelectDropdown passes correct props to OptionContent: OptionContent gets correct props when an option is not selected 1`] = `
Object {
  "$disabled": undefined,
  "$error": false,
  "$isHighlighted": undefined,
  "$isLoading": false,
  "$multi": false,
  "$required": false,
  "$searchable": true,
  "$selected": false,
  "$size": "default",
  "$type": "select",
  "$width": 100,
  "aria-readonly": undefined,
  "aria-selected": false,
  "children": <span>
    option.label
  </span>,
}
`;

exports[`SelectDropdown renders StatefulMenu: Passes correct overrides to the StatefulMenu 1`] = `
Object {
  "List": Object {
    "component": Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "StyledList",
      "render": [MockFunction],
    },
    "props": Object {
      "$maxHeight": "1000px",
      "aria-multiselectable": false,
    },
    "style": [Function],
  },
  "Option": Object {
    "props": Object {
      "getItemLabel": [Function],
      "onMouseDown": [Function],
      "overrides": Object {
        "ListItem": Object {
          "component": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "StyledListItem",
            "render": [MockFunction],
          },
          "props": Object {
            "role": "option",
          },
          "style": undefined,
        },
      },
    },
  },
}
`;
